### ISSUE TITLE
Idea. Forgot password

### ISSUE THREAD

**Issue by jaegeral** [MISP/MISP#1799]
Now with more and more organisations using MISP the number of accounts (per instance) is increasing, so managing the accounts is taking more and more time. One thing that is a regular issue is password reset so it might be handy to have a self service for **forgot password**[1].

PGP key for users should / can be stored in advance so the password reset process can be used as it is used for administrators to set initial password.

Or was there a good reason for not having a "forgot password"-feature? - If so, we should mention it at a place that people can find, as I was not able to find that anywhere in the MISP docs.

Opinions?

[1] https://www.owasp.org/index.php/Forgot_Password_Cheat_Sheet

**Comment by adulau** [comment]
We had some discussions about it and especially regarding the specific security aspect like potential pre-auth vulnerabilities, DoS (locking of accounts), side-channel/timing attacks and so on.

We might introduce a functionality of password reset for the users having a PGP key or an S/MIME key. The idea is to put this functionality in an async process with a dedicated worker to limit risk of side-channel attacks and local DoS.

Will this cover your use-cases?

**Comment by jaegeral** [comment]
Well DoS-risk could be reduced by Max N passwords resets per source IP per timeframe

But yes, it would be okay for me for PGP / SMIME users online.

Maybe it is also okay to make password reset for users as an opt-in module for admins to decide, so they can make a risk evaluation based on their scenario. E.g. if your MISP instance with the holy IOCs is segregated and Source IP filtered etc the risk for DOS, sidechannel attacks etc is lot lower then running an MISP on the public net.

**Comment by cristianbell** [comment]
This all sounds good, but what about for users without a PGP or S/MIME key? 
Maybe a message to the site admin or orgadmin saying user xyz requests a password reset? 

Of course protecting it as the sign in with SecureAuth.amount and SecureAuth.expire, and maybe also a capcha check, timer for requesting a password reset,...

**Comment by jaegeral** [comment]
@cristianbell 
Agree, if a User has no PGP / SMIME a mail to the admin would be a good idea.
